### ISSUE TITLE
[strict-boolean-expressions] Add option to allow `null` and `undefined`

### ISSUE THREAD

**Issue by sindresorhus** [typescript-eslint/typescript-eslint#698]
The `strict-boolean-expressions` rule was added recently, but it's missing many of the options from the [TSLint rule](https://palantir.github.io/tslint/rules/strict-boolean-expressions/). I agree that many of those options makes the rule pointless, but I also think the rule is too strict by default. It results in very verbose code for no good reason.

An option to allow `undefined` and `null` in the union would be useful. Equivalent to the TSLint `allow-null-union` and `allow-undefined-union` options.

Such option would allow this pattern which is very common:

```ts
if (previousComment && previousComment.closest('.js-timeline-item') !== comment) {
```

^ Which does a quick null/undefined check before operating on a variable.

**Comment by Djaler** [comment]
I really want configuration for this rule. Restrictions for numbers and enums are awesome, but I usually check for empty string by implicit cast

**Comment by bradzacher** [comment]
This is where I wish that typescript would implement the null coalesce operator.
It's one of the few things I miss when I jump from the flow codebase at work to my typescript side projects.

This is just so much nicer.
```js
if (previousComment?.closest('.js-timeline-item') !== comment) {
```

**Comment by glen-84** [comment]
> This is where I wish that typescript would implement the null coalesce operator.

This is [optional chaining](https://github.com/tc39/proposal-optional-chaining) / safe navigation / null propagation.

TypeScript will only implement it once it reaches stage 3 or 4 (I can't remember which it is), but the good news is that it recently moved to stage 2, so progress is being made.

There's also a [Nullish coalescing proposal](https://github.com/tc39/proposal-nullish-coalescing) at stage 3.

**Comment by bradzacher** [comment]
Yeah I mixed up the names sorry.

Stage 3 is when typescript looks to implement. They're looking to implement the coalesce operator now.
